### PR TITLE
maintainers: add a helper script for the options doc conversion

### PR DESCRIPTION
**Pull request by pennae**
this script can be used to attempt an automatic conversion of option
docs for most modules. it'll also show a diff of options.json before and
after the changes, which should be a good form for checking for unwanted
changes. we specifically show a json diff rather than an xml diff
because newline changes in json are "\n" added are removed, and those
are easier for diff tools to pick out and show in a meaningful way for
this process.

it does *not* check for incorrectly applied changes though, those aren't
easy enough to do automatically for this script.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/scripts/mdize-module.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -I nixpkgs=. -i bash -p delta jq perl
+
+cat <<'EOF'
+This script attempts to automatically convert option descriptions from
+DocBook syntax to markdown. Naturally this process is incomplete and
+imperfect, so any changes generated by this script MUST be reviewed.
+
+
```
Comment (roberth) [inline maintainers/scripts/mdize-module.sh:9]: Would it make sense to add a non-exhaustive list of known issues to be aware of?

File: maintainers/scripts/mdize-module.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+echo "Building options.json again ..."
+new_options=$(build-options-json)
+
+
+! cmp -s {$old_options,$new_options}/share/doc/nixos/options.json && {
+    diff -U10 \
+        <(jq . <$old_options/share/doc/nixos/options.json) \
+        <(jq . <$new_options/share/doc/nixos/options.json) \
+        | delta
```
Comment (roberth) [inline maintainers/scripts/mdize-module.sh:68]: I'm getting a diff from this where it's docbook in red and markdown in green. I *think* what's supposed to happen is that the new options are mapped back to docbook by `mergeJSON.py` for the purpose of the comparison.

My invocation was

    ./maintainers/scripts/mdize-module.sh nixos/modules/virtualisation/qemu-vm.nix

Comment (pennae) [reply maintainers/scripts/mdize-module.sh:68]: yeah, turns out i'm an idiot and merged the regex that applied `mdDoc` into the big perl call incorrectly :facepalm: now we get `inline_html` md blocks for `qemu-vm.nix`, which also isn't great. needs work.

Comment (pennae) [reply maintainers/scripts/mdize-module.sh:78]: on second though the doc build crashing is probably a good thing. that's probably the easiest way of spotting xml in markdown docs :)

**Review by roberth — APPROVED** [review]

**Pull request merged**